docs/source/installation.rst

```diff
@@ -0,0 +1,147 @@
+    sudo locale-gen en_US en_US.UTF-8 
+    sudo update-locale LC_ALL=en_US.UTF-8 LANG=en_US.UTF-8 export LANG=en_US.UTF-8
+    sudo apt install software-properties-common sudo add-apt-repository universe
+    sudo apt update && sudo apt install curl gnupg lsb-release sudo curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg
+    echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/ros2.list > /dev/null
+    sudo apt install ros-galactic-desktop
+    
+
+
+Step 2 - Get Terminator
+-----------------------
+We primarily use terminator instead of the default terminal since it allows for both horizontal and vertical splitting (which obviously increases overall productivity). Start up the default terminal and type:
+
+``sudo apt install terminator``
+
+Step 3 - Source Environment Variables
+-------------------------------------
+For the time being, we do not recommend placing these commands into ~/.bashrc (or equivalent) because we will be switching between ROS and ROS2. Instead, you will need to source your environment with each shell instance.
+
+``source /opt/ros/galactic/setup.bash``
+
+
+
+Setup Workspace
+===============
+
+Step 4 - Make File Structure
+----------------------------
+.. code-block::
+
+    mkdir -p ~/roboracing_ws/src
+    cd ~/roboracing_ws/src
+    git clone https://github.com/RoboJackets/roboracing-software.git -b ros2/main --recursive
+
+Step 5 - Install Dependencies
+-----------------------------
+
+.. code-block::
+
+    cd ~/roboracing_ws 
+    sudo apt install -y python3-colcon-common-extensions python3-rosdep 
+    rosdep install --from-path src --ignore-src -y
+
+Step 6 - Building the Code
+--------------------------
+
+.. code-block::
+
+    colcon build 
+    source ~/roboracing_ws/install/setup.bash
+
+Similar to step 3, you should run ``source ~/roboracing_ws/install/setup.bash`` with every new shell instance.
+
+Documentation
+=============
+
+Step 1 - Install Sphinx
+-----------------------
+
+.. code-block::
+
+    sudo apt update
+    sudo apt install python3-sphinx
+    pip install furo
+
+Step 2 - Install PlantUML
+-------------------------
+
+.. code-block::
+
+    sudo apt install default-jre
+    sudo apt install graphviz
+    pip install sphinxcontrib-plantuml
+
+Download most recent version of plantuml from: https://github.com/plantuml/plantuml/releases/latest.
+Download the one that has no suffix, meaning NOT -javadoc or -sources.
+
+Download the file into ``~/java/`` and rename the file to plantuml.jar. The download path must be: ``~/java/plantuml.jar``
+
+Optional:
+
+To enable plantuml from anywhere in your file structure add the following line to your bashrc. :doc:`linux_info/bashrc`
+
+.. code-block::
+
+    echo alias plantuml="'java -jar /home/<username>/java/plantuml.jar'" >> ~/.bashrc
```

Comment:
Why does it say to use /home/<username> instead of ~/? Not necessary to change I'm just curious. Though maybe we should change ~/java with /home/<username> because some people might not know what ~ means

Reply:
I will write a note explaining what ~ is